crates/sui-package-management/src/lib.rs

```diff
@@ -140,45 +141,42 @@ pub fn resolve_published_id(
 ) -> Result<ObjectID, PublishedAtError> {
     // Look up a valid `published-at` in the `Move.toml` first, which we'll
     // return if the Move.lock does not manage addresses.
-    let published_id_in_manifest = match published_at_property(package) {
-        Ok(v) => Some(v),
-        Err(PublishedAtError::NotPresent) => None,
-        Err(e) => return Err(e), // An existing but invalid `published-at` in `Move.toml` should fail early.
-    };
+    let published_id_in_manifest = manifest_published_at(package);
+
+    match published_id_in_manifest {
+        Ok(_) | Err(PublishedAtError::NotPresent) => { /* nop */ }
```

Comment:
I feel like this manifest_published_at should be a different function, that checks if it is present or not. Somehow (as I don't know this code at all), it gets confusing that if the function's result is Ok, an error is returned.